### PR TITLE
feat: add `getMessageByNonce` to `Provider`

### LINKED ISSUE #1538: Implement `getMessageByNonce` or similar
We probably need to request this feature from `fuel-core.`

> [!NOTE]
> - Incoming from https://github.com/FuelLabs/fuels-ts/discussions/1537

### PR DESCRIPTION
**Pull request by Dhaiwat10**
## Summary

This PR adds a new method called `getMessageByNonce` to the `Provider` class which lets users retrieve a message using its nonce.

Closes #1538

### CODE REVIEW

File: packages/account/src/providers/provider.ts [hunk 1]
```diff
@@ -1635,4 +1636,21 @@ export default class Provider {
...
     return new TransactionResponse(transactionId, this);
   }
+
+  /**
+   * Returns Message for given nonce.
+   *
+   * @param nonce - The nonce of the message to retrieve.
+   * @returns A promise that resolves to the Message object.
+   */
+  async getMessageByNonce(nonce: string): Promise<GqlMessage | undefined> {
```
[the remark this answers is not in the document]
Comment (Dhaiwat10) [reply packages/account/src/providers/provider.ts:1646]: cc @maschad

[the remark this answers is not in the document]
Comment (Dhaiwat10) [reply packages/account/src/providers/provider.ts:1646]: fwiw I agree with your point @petertonysmith94

[the remark this answers is not in the document]
Comment (maschad) [reply packages/account/src/providers/provider.ts:1646]: I agree with @petertonysmith94 as well, `null` seems more applicable. Thanks for referencing ECMA standards as well @petertonysmith94 very good point of information.

**Review by maschad — APPROVED** [review]

**Review by arboleya — APPROVED** [review]
Nicely done! :shipit:

File: .changeset/blue-plants-mate.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@fuel-ts/account": patch
+---
+
+feat: add `getMessageByNonce` to `Provider`
```
Comment (arboleya) [inline .changeset/blue-plants-mate.md:5]: I missed this initially, but it should've been a patch because it doesn't break anything.

**Pull request merged**